src/main/java/org/gradle/profiler/buildops/BuildOperationInstrumentation.java

```diff
@@ -20,15 +19,15 @@
 
 public class BuildOperationInstrumentation extends GradleInstrumentation {
     private final File configurationTimeDataFile;
-    private final ImmutableMap<String, File> buildOperationDataFiles;
+    private final Map<String, File> buildOperationDataFiles;
     private final boolean measureConfigTime;
 
     public BuildOperationInstrumentation(boolean measureConfigTime, List<String> measuredBuildOperations) throws IOException {
         this.measureConfigTime = measureConfigTime;
         this.configurationTimeDataFile = File.createTempFile("gradle-profiler", "build-ops-config-time");
         this.configurationTimeDataFile.deleteOnExit();
         this.buildOperationDataFiles = measuredBuildOperations.stream()
-            .collect(ImmutableMap.toImmutableMap(Function.identity(), BuildOperationInstrumentation::createBuildOperationTempFile));
+            .collect(Collectors.toMap(Function.identity(), BuildOperationInstrumentation::createBuildOperationTempFile));
```

Comment:
You can use `ImmutableMap.toImmutableMap()` here to keep this as an `ImmutableMap`.

Reply:
I can't, since on the `gradle/gradle` side we use the android variant of guava, which does not contain the method. That is why I switched to using a regular map instead of an immutable one.

Reply:
Right. And we need to access the profiler in an embedded way?

Reply:
Yeah, I want to use the Java API instead of spawning yet another process and parsing the CSV result.